### PR TITLE
Servlet for waypoints

### PR DESCRIPTION
**Pull request by dianapvalles**
- Add servlet to get the waypoint for its corresponding grid
- Had to delete txt file since the path was not being read unless it was hardcoded

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java [hunk 1]
```diff
@@ -1,46 +1,148 @@
...
+@WebServlet("/grids")
+public class WaypointsLoader extends HttpServlet {
   private static final int NUM_GRIDS_ROWS = 6;
   private static final int NUM_GRIDS_COLS = 16;  
-  private static final HashMap<Grid, Location> MAP = readWaypoints();
+  private static final Gson gson = new Gson();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    HashMap<Grid, Location> MAP = readWaypoints();
```
Comment (wait321) [inline portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java:22]: private static final
you're recreating this every time this method is called which is not necessary

File: portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java [hunk 2]
```diff
@@ -1,46 +1,148 @@
...
+import javax.servlet.http.HttpServletResponse;
 
-public class WaypointsLoader {
+/** Handles fetching and saving grids data. */
+@WebServlet("/grids")
+public class WaypointsLoader extends HttpServlet {
   private static final int NUM_GRIDS_ROWS = 6;
   private static final int NUM_GRIDS_COLS = 16;  
-  private static final HashMap<Grid, Location> MAP = readWaypoints();
+  private static final Gson gson = new Gson();
```
Comment (wait321) [inline portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java:18]: `gson` => `GSON` it's a constant

File: portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java [hunk 3]
```diff
@@ -1,46 +1,148 @@
...
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException{
+    HashMap<Grid, Location> MAP = readWaypoints();
+    String json = gson.toJson(MAP);
```
Comment (wait321) [inline portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java:23]: don't pass back the whole map. the request has grid row and col. just return MAP.get(new Grid(requestRow, requestCol))

File: portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java [hunk 4]
```diff
@@ -1,46 +1,148 @@
...
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+  }
   
   public static HashMap<Grid,Location> readWaypoints(){
     HashMap<Grid,Location> map = new HashMap<>();
+    double[] coordinatesWaypoint = {31.675332,-106.44099,
```
Comment (wait321) [inline portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java:30]: `private static final double[] COORDINATES_WAYPOINT`

File: portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java (hunk 1 above)
Comment (dianapvalles) [reply portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java:22]: Noted, thank you!

File: portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java (hunk 2 above)
Comment (dianapvalles) [reply portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java:18]: Noted, thanks!

File: portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java (hunk 4 above)
Comment (dianapvalles) [reply portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java:30]: Noted, thank you!

File: portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java (hunk 3 above)
Comment (dianapvalles) [reply portfolio/src/main/java/com/google/sps/servlets/WaypointsLoader.java:23]: Noted, thanks!

**Pull request merged**